packages/core/src/api/serialization/serialization.test.ts

```diff
@@ -0,0 +1,296 @@
+import { afterEach, beforeEach, describe, expect, it } from "vitest";
+import { BlockNoteEditor } from "../../BlockNoteEditor";
+import { Editor } from "@tiptap/core";
+import { customBlockSerializer } from "./serialization";
+import { uploadToTmpFilesDotOrg_DEV_ONLY } from "../../extensions/Blocks/nodes/BlockContent/ImageBlockContent/uploadToTmpFilesDotOrg_DEV_ONLY";
+import { createBlockSpec } from "../../extensions/Blocks/api/block";
+import {
+  imagePropSchema,
+  renderImage,
+} from "../../extensions/Blocks/nodes/BlockContent/ImageBlockContent/ImageBlockContent";
+import { defaultBlockSchema } from "../../extensions/Blocks/api/defaultBlocks";
+import { cleanHTML } from "../formatConversions/formatConversions";
+import { DOMSerializer, Fragment, Node } from "prosemirror-model";
+
+// This is a modified version of the default image block that does not implement
+// a `serialize` function. It's used to test if the custom serializer by default
+// serializes custom blocks using their `render` function.
+const SimpleImage = createBlockSpec({
+  type: "simpleImage",
+  propSchema: imagePropSchema,
+  containsInlineContent: false,
+  render: renderImage as any,
+});
+
+const customSchema = {
+  ...defaultBlockSchema,
+  simpleImage: SimpleImage,
+};
+
+let editor: BlockNoteEditor<typeof customSchema>;
+let tt: Editor;
+
+beforeEach(() => {
+  (window as Window & { __TEST_OPTIONS?: any }).__TEST_OPTIONS = {};
+
+  editor = new BlockNoteEditor({
+    blockSchema: customSchema,
+    uploadFile: uploadToTmpFilesDotOrg_DEV_ONLY,
+  });
+  tt = editor._tiptapEditor;
+});
+
+afterEach(() => {
+  tt.destroy();
+  editor = undefined as any;
+  tt = undefined as any;
+
+  delete (window as Window & { __TEST_OPTIONS?: any }).__TEST_OPTIONS;
+});
+
+// Tests `serializeNode`, `serializeFragment`, and `cleanHTML` methods.
+async function serializeAndCompareSnapshots(
+  serializer: DOMSerializer,
+  blockGroup: Node
+) {
+  const serializeNodeHTML = (
+    serializer.serializeNode(blockGroup) as HTMLElement
+  ).outerHTML;
+  const serializeFragmentHTML = (
+    serializer.serializeFragment(Fragment.from(blockGroup)) as HTMLElement
+  ).firstElementChild!.outerHTML;
+
+  expect(serializeNodeHTML).toMatchSnapshot();
```

Comment:
should `serializeNodeHTML` and `serializeFragmentHTML` be the same? In that case, let's only store 1 snapshot, and just compare them to eachother